File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    citations.sort()
    #[0,1,3,5,6]
    answer = len(citations)
    ind = 0
    while ind<len(citations):
        if citations[ind] >= answer:
            break
        else:
            ind +=1
            answer -=1
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.kt
class Solution {
    fun solution(citations: IntArray): Int {
        var answer = 0
        val maxdata = citations.maxOrNull() ?:0
        for (i in maxdata downTo 0 ) {
            val count1 = citations.count{it >= i}
            val count2 = citations.count{it <= i}
            if (count1 >= i && count2 <= i) {
                answer = i
                break
            }
        }
        return answer
    }
}